### ISSUE TITLE
2_Image_Restoration_deraing_raindrop_noise1

### ISSUE THREAD

**Issue by sunqiqi123** [nachifur/RDDM#39]
在”2_Image_Restoration_deraing_raindrop_noise1“文件中。图像恢复——去雨训练
在residual_denoising_diffusion_pytorch.py的UnetRes中，发现：
 out_res = x
            for block1, block2, attn, upsample in self.ups_no_skip:
                out_res = block1(out_res, t)
                out_res = block2(out_res, t)
                out_res = attn(out_res)

                out_res = upsample(out_res)

            out_res = self.final_res_block_1(out_res, t)
            out_res = self.final_conv_1(out_res)

            for block1, block2, attn, upsample in self.ups:
                x = torch.cat((x, h.pop()), dim=1)
                x = block1(x, t)

                x = torch.cat((x, h.pop()), dim=1)
                x = block2(x, t)
                x = attn(x)

                x = upsample(x)

            x = torch.cat((x, r), dim=1)
            x = self.final_res_block_2(x, t)
            out_res_add_noise = self.final_conv_2(x)

            return out_res, out_res_add_noise    
想咨询您关于为什么out_res用的是self.ups_no_skip，而跟 out_res_add_noise不一样。而在“0_Partially_path-independent_generation”文件中，使用两个unet训练时，res和noise用的是相同的unet结构。对此，我有一些疑惑。

**Comment by nachifur** [comment]
https://github.com/nachifur/RDDM/blob/461957d0e9b22e4ada7daceb75ae7349202eea85/experiments/2_Image_Restoration_deraing_raindrop_noise1/train.py#L78

**Comment by nachifur** [comment]
share_encoder = 1可以对应这块
![image](https://github.com/user-attachments/assets/0f5b2234-ee8d-4fdc-b89b-b2ac98ff34d6)
